tables/TaQL/TaQLNodeDer.cc

```diff
@@ -1667,7 +1667,7 @@ void TaQLRecFldNodeRep::show (std::ostream& os) const
     os << itsFromName;
   } else if (itsValues.isValid()) {
     if (itsValues.nodeType() == TaQLNode_Multi  &&
-        ((const TaQLMultiNodeRep*)(itsValues.getRep()))->itsNodes.empty()) {
+        dynamic_cast<const TaQLMultiNodeRep*>(itsValues.getRep())->itsNodes.empty()) {
```

Comment:
```
dynamic_cast<const TaQLMultiNodeRep&>(*itsValues.getRep()).itsNodes.empty()) {
```
This way when the cast fails it thows an exception instead of having undefined behaviour due to deferring a `nullptr`.

Reply:
When failing, the dynamic cast of a pointer returns a null pointer and does not throw. It will result in a core dump. A reference will throw though, so I'll use that.